### PR TITLE
Fix examples to point to lts-11.1 and update docs

### PR DESCRIPTION
**Pull request by harshavardhana**

### CODE REVIEW

**Comment by harshavardhana** [comment]
This additionally also fixes a major OOM bug in putObject implementation for single stream upload. @donatello @krisis - Thanks for @krisis for helping.

**Comment by harshavardhana** [comment]
Fixed the unused values in this PR @donatello @krisis

File: docs/API.md [hunk 1]
```diff
@@ -323,10 +339,9 @@ main = do
 
   -- Performs a recursive listing of all objects under bucket "test"
   -- on play.minio.io.
-  res <- runMinio minioPlayCI $ do
-    listObjectsV1 bucket Nothing True $$ sinkList
+  res <- runMinio minioPlayCI $
+    runConduit $ listObjects bucket Nothing True .| mapM_C (\v -> (liftIO $ print v))
```
Comment (krisis) [inline docs/API.md]: This should be `listObjectsV1`.

File: src/Network/Minio/PutObject.hs [hunk 2]
```diff
@@ -68,7 +69,7 @@ putObjectInternal b o opts (ODStream src sizeMay) = do
     -- got file size, so check for single/multipart upload
     Just size ->
       if | size <= 64 * oneMiB -> do
-             bs <- C.runConduit $ src C..| CB.sinkLbs
+             bs <- C.runConduit $ src C..| takeC (fromIntegral size) C..| CB.sinkLbs
```
Comment (krisis) [inline src/Network/Minio/PutObject.hs]: Good catch!

**Pull request merged**